### PR TITLE
feat: Add aria-current attribute on next/link

### LINKED ISSUE #6014: Set aria-current attributes on next/link
# Feature request

## Is your feature request related to a problem? Please describe.
Right now you have to do this yourself and I'd bet many forget this. It would be really nice to have this included in `next/link` by default.

## Describe the solution you'd like
Set a `aria-current` attribute on the active link(s). I could work on this, but I might need some guidance on how this should be implemented since we probably have to wrap the link component in a `withRoute` hoc.

## Describe alternatives you've considered
N/A

## Additional context
https://www.w3.org/TR/wai-aria-1.1/#aria-current

### PR DESCRIPTION
**Pull request by alexandernanberg**
Fixes #6014 

Might be relevant to add some tests for this also

### CODE REVIEW

File: packages/next/client/link.js [hunk 1]
```diff
@@ -144,6 +144,11 @@ class Link extends Component {
       props.href = _Router._rewriteUrlForNextExport(props.href)
     }
 
+    const isCurrent = props.href === router.pathname
```
Comment (jaydenseric) [inline packages/next/client/link.js]: Should this be:

```js
const isCurrent = props.href === router.asPath
```

**Comment by rashidul0405** [comment]
Framework should not do everything and we should keep the core lightweight. Usually most of the apps add some sort of class for active links, this should be the responsibility  of the developer?

File: packages/next/client/link.js (hunk 1 above)
Comment (alexandernanberg) [reply packages/next/client/link.js]: Yeah probably, wasn't aware of the `asPath` property 👍

**Comment by alexandernanberg** [comment]
@rashidul0405 I do agree that the core should be lightweight but I don't think that applies to this change though. The `aria-current` property is not a substitute for classes on active links (although it can be used as one in some cases), this is mainly for people using screen readers to indicate which nav link is the active/current one. 

Imo it makes sense for this to be included because it doesn't really add any "weight" and would make sites using next.js more accessible by default.

**Comment by timneutkens** [comment]
I'm going to close this in favor of something we're thinking of internally.

**Comment by lachlanjc** [comment]
> I'm going to close this in favor of something we're thinking of internally. More 🔜

Any update here @timneutkens?

**Comment by stefanprobst** [comment]
> I'm going to close this in favor of something we're thinking of internally. More soon

@timneutkens is there any publc info on what you're proposing wrt `aria-current` on nav links?

**Pull request closed without merging**